hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -322,7 +327,15 @@ private void transitionState(HoodieInstant fromInstant, HoodieInstant toInstant,
         ValidationUtils.checkArgument(metaClient.getFs().exists(new Path(metaClient.getMetaPath(),
             fromInstant.getFileName())));
         // Use Write Once to create Target File
-        createImmutableFileInPath(new Path(metaClient.getMetaPath(), toInstant.getFileName()), data);
+        if (allowRedundantTransitions) {
+          createFileInPath(new Path(metaClient.getMetaPath(), toInstant.getFileName()), data);
+        } else {
+          if (allowRedundantTransitions) {
+            createFileInPath(new Path(metaClient.getMetaPath(), toInstant.getFileName()), data);
```

Comment:
can be removed?

Reply:
Thanks for catching it. It was a bad merge.